spec/habitat_spec.cr

```diff
@@ -52,6 +52,19 @@ end
 class Settings::Index < Parent
 end
 
+# Test that when we set the default value to a setting, and the value
+class ConfigWithDefaultException
+  # Uncomment this to see a MissingSettingError
+  # Ref: https://github.com/luckyframework/habitat/issues/46
+  # Habitat.create do
+  #  setting explode : String = ConfigWithDefaultException.blows_up
+  # end
```

Comment:
I wasn't sure how to test this since this throws the exception right at runtime. If you uncomment these 3 lines and run the specs, you'll see a MissingSettingError thrown. Before this PR it would throw an exception related to recurrsion.